dcrwallet.go

```diff
@@ -289,7 +290,13 @@ func rpcClientConnectLoop(legacyRPCServer *legacyrpc.Server, loader *wallet.Load
 				legacyRPCServer.SetChainServer(chainClient)
 			}
 			if cfg.EnableTicketBuyer {
-				startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				pm, err = startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				if err != nil {
+					log.Errorf("Unable to start ticket buyer: %v", err)
+					return
+				}
+				pm.Start()
+				loader.SetWaitFn(pm.WaitForShutdown)
```

Comment:
Yup, tested and found that calling `CloseWallet` hangs as expected, because `pm.Stop` needs to be called first, since we're waiting on `pm.WaitForShutdown`. Thanks for catching that.

Updated to add `pm.Stop` to `SetWaitFn` as well. This makes sure we stop and wait on the purchase manager whenever wallet is unloaded.